f8a_worker/workers/init_analysis_flow.py

```diff
@@ -104,6 +105,8 @@ def execute(self, arguments):
         finally:
             # always clean up cache
             shutil.rmtree(cache_path)
+            if arguments['ecosystem'] == "npm":
```

Comment:
I just gone through the flow, I feel like instead of being specific to ecosystem, deleting `source_tarball_path` should do the needful. Also I see that `IndianaJones.fetch_artifact` called in couple of places in `init_analysis_flow.py` for maven ecosystem, should we clear those as well?

Reply:
tar balls are stored in s3 not in local memory. They are extracted and worked upon. 
As for the indiana jones call, there are a few weird steps everywhere during the ingestion procedure. The call here we need to take is, do we make any changes in a working stable platform? My suggestion at the moment would be 'no'.

Reply:
`source_tarball_path` variable holds a path to tarball which has to be put into S3. So deleting `source_tarball_path` after putting into S3 is as same as deleting `~/.npm`. Probable you can check that in your devcluster if you have the setup.

Reply:
sample: source_tarball_path -> /var/lib/f8a_worker/worker_data/tmpbrq6er3j/package.tgz location where we store the tarball. And then after processing, we delete all the contents which is inside /var/lib/f8a_worker/worker_data . So in the end, we dont have any tarball etc in our disk or local memory